pkg/controller/gc.go

```diff
@@ -572,3 +576,34 @@ func (c *Controller) isOVNProvided(providerName string, pod *corev1.Pod) (bool,
 	}
 	return true, nil
 }
+
+func (c *Controller) getVmLsps() []string {
+	var vmLsps []string
+
+	if !c.config.EnableKeepVmIP {
+		return vmLsps
+	}
+
+	nss, err := c.namespacesLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("failed to list namespaces, %v", err)
+		return vmLsps
+	}
+
+	for _, ns := range nss {
+		vms, err := c.config.KubevirtClient.VirtualMachine(ns.Name).List(&metav1.ListOptions{})
+		if err != nil {
+			if !k8serrors.IsNotFound(err) {
+				klog.Errorf("failed to list vm in namespace %s, %v", ns, err)
+			}
+			continue
+		} else {
+			for _, vm := range vms.Items {
+				vmLsp := ovs.PodNameToPortName(vm.Name, ns.Name, util.OvnProvider)
+				vmLsps = append(vmLsps, vmLsp)
+			}
```

Comment:
What if the VM has multiple LSP? Is it possible to get all LSPs?

Reply:
1、Kubevirt can be configured multiple lsps with multi-cni.
2、The pod does not exist, only lsp for vm pod exists. So there is no way to get attachnetworks by pod.
3、It's not worth it to check every lsp and make sure it is used for vm pod.
So the default provider can satisfy common situations